src/main/java/ratpack/zipkin/internal/RatpackServerClientLocalSpanState.java

```diff
@@ -37,24 +37,29 @@
   private final MDCProxy mdc;
   private final Endpoint endpoint;
 
-  public RatpackServerClientLocalSpanState(final String serviceName, int ip, int port) {
-    this(serviceName, ip, port, Execution::current, new DefaultMDCProxyImpl());
-  }
-
-  RatpackServerClientLocalSpanState(final String serviceName,
-                                    int ip,
-                                    int port,
-                                    final Supplier<MutableRegistry> registry,
-                                    final MDCProxy mdc) {
+  public RatpackServerClientLocalSpanState(final Endpoint endpoint,
+                                           final Supplier<MutableRegistry> registry,
+                                           final MDCProxy mdc) {
     this.registry = registry;
-    this.endpoint = Endpoint.builder()
-                            .serviceName(serviceName)
-                            .ipv4(ip)
-                            .port(port)
-                            .build();
+    this.endpoint = endpoint;
     this.mdc = mdc;
   }
 
+  public RatpackServerClientLocalSpanState(final Endpoint endpoint) {
```

Comment:
@llinder I think the `RatpackServerClientLocalSpanState(Endpoint, Supplier<MutableRegistry>, MDCProxy)` constructor is just used [here](https://github.com/edwardsoo-ss/ratpack-zipkin/blob/de80cbf2ad5551ac1b74f0bbb6fa2cac8ccbb901/src/test/groovy/ratpack/zipkin/internal/RatpackServiceClientLocalSpanStateSpec.groovy#L33) in a test setup. Could make it package-scoped.